src/views/Setup.vue

```diff
@@ -1,10 +1,33 @@
 <script setup>
-import { ref } from 'vue';
+import { ref, computed, watch } from 'vue';
 import { useRouter } from 'vue-router';
+import { useEns } from '@/composables/useEns';
+import { useWeb3 } from '@/composables/useWeb3';
+
+const { web3 } = useWeb3();
+const web3Account = computed(() => web3.value.account);
 
 const router = useRouter();
+const { getEnsNames } = useEns();
 
 const id = ref('');
+const loading = ref(false);
+const domains = ref(null);
+
+function applyEns(ens) {
+  id.value = ens;
+}
+
+function loadEns() {
+  loading.value = true;
+  getEnsNames()
+    .then(res => {
+      domains.value = res.account?.domains.slice(0, 5);
+    })
+    .finally(() => (loading.value = false));
+}
+
+watch(web3Account, loadEns);
```

Comment:
```suggestion
loadEns();
watch(web3Account, loadEns);
```

To make sure ENS names are loaded when entering the setup page and not ONLY when web3Account changes.